Add basic items

JeanJoeris opened the pull request:
Built minimum functionality to satisfy the user story

Users are able to visit an items index and see existing items. Currently it is just showing item title

JeanJoeris committed:
add feature test for user seeing items, along with test gems

JeanJoeris committed:
finish writing test and further update gem file

JeanJoeris committed:
add model and migration for items

JeanJoeris committed:
add route to items index and controller with index action

JeanJoeris committed:
add items index view and fixed test typo

JeanJoeris committed:
redo migration and test to be closer in line with client user story

JeanJoeris commented:
The card said items had title, description, price and image. Currently the index is only showing name. What do we want the index to show?

Also, thoughts on image as a path?

rsbarbo commented:
@JeanJoeris @j-sm-n - attached to the card Sal gave us the following hint:

  As a visitor
  When I visit "/items"
  I can see all existing items

So I would think for now we need to display all the info for each item. Maybe we can create thumbnails to each product, and have:
Title 
in the box image 
then description
underneath price 

I would gladly brainstorm, but that would be my take for now. 

Also amazing job Jean on putting those tests together.  Thank you so much 
Rapha Barbo

JeanJoeris committed:
add description and price to view

j-sm-n commented on db/migrate/20160913023501_create_items.rb:
```diff
@@ -0,0 +1,11 @@
+      t.string :description
+      t.integer :price
+      t.string :image_path
+      t.timestamp null: false
```
I think this line needs to be t.timestampS. The `created_at` and `updated_at` doesn't look like they got created in the schema.

rsbarbo replied:
@j-sm-n Good catch 👍

JeanJoeris replied:
> I think this line needs to be t.timestampS. The created_at and updated_at doesn't look like they got created in the schema.

Thanks for the catch

j-sm-n commented:
@JeanJoeris @rsbarbo To continue off the topic of what information to display on the items index page, what if we had a short description field and a long description field for each item? That way, we could display the short one for the index page and then the longer description for the item show page? Just a thought :)

But this looks great so far, Jean! 🍰

JeanJoeris committed:
add item image to the items index

JeanJoeris committed:
redo migration to fix timestamps typo

JeanJoeris committed:
add item model testing for validating presence

JeanJoeris committed:
update item to have an optional longer description, and add model test to verify this

JeanJoeris commented:
I fixed the timestamps, added image to index and corresponding test update. Added a model testing verifying presence title, description, price and image_path.

I also added a long_description, though it is not being validated for presence right now

j-sm-n commented:
Nice @JeanJoeris! Thanks for adding that `long_description` attribute :)

After our standup this afternoon, let's review together and merge.

JeanJoeris committed:
change to single quotes when possible and shorten lines, all to fix hound violations

change to single quote in user sees items feature test

fix spacing and hound violations

rsbarbo commented:
@j-sm-n @JeanJoeris  everything looks good for our first merge. I will go ahead and merge this pull request, and continue to work on the additional cards. Thanks Rapha Barbo
